RenderStateNotation/src/RenderStateNotationParserImpl.cpp

```diff
@@ -476,33 +476,58 @@ Bool RenderStateNotationParserImpl::ParseString(const Char* StrData, Uint32 Leng
     return true;
 }
 
-const PipelineStateNotation* RenderStateNotationParserImpl::GetPipelineStateByName(const Char* Name) const
+const PipelineStateNotation* RenderStateNotationParserImpl::GetPipelineStateByName(const Char* Name, PIPELINE_TYPE PipelineType) const
 {
-    auto Iter = m_PipelineStateNames.find(Name);
-    if (Iter != m_PipelineStateNames.end())
-        return &m_PipelineStates[Iter->second].get();
+    auto FindPipeline = [this](const Char* Name, PIPELINE_TYPE PipelineType) -> const PipelineStateNotation* //
+    {
+        const auto Iter = m_PipelineStateNames.find(std::make_pair(HashMapStringKey{Name, true}, PipelineType));
```

Comment:
`true` in `HashMapStringKey{Name, true}` means that `HashMapStringKey` will make a copy of the string. This is necessary when inserting an element into the map. When searching, however, making a copy is not necessary because `Name` will be valid while the function is running.